### DISCUSSION TITLE
How does one handle authentication, authorization and sessions?

### DISCUSSION THREAD

**Discussion by metaperl** [elimintz/justpy#386] [Q&A]
JustPy appears to be self-contained --- not designed to work with a traditional web app framework. As such, one wonders how to handle traditional web application tasks such as authentication, authorization and sessions.

**Comment by elimintz** [comment]
Justpy comes with session support:
https://justpy.io/tutorial/sessions/

**Comment by sandeep-gh** [comment]
I am looking at these issues currently. Seems to me that we need to add authentication middleware in addition to sessions. 
Starlette [docs](https://www.starlette.io/authentication/) has outline for BasicAuthBackend. With auth middleware plugged in you can use requires decorater to allow endpoint access to authenticated users : `@requires('authenticated')`
